src/lineup/ARankingView.ts

```diff
@@ -287,8 +287,21 @@ export abstract class ARankingView extends AView {
     colDesc.color = this.colors.getColumnColor(id);
     const accessor = createAccessor(colDesc);
 
+    // find last index of the current selected ID
+    const lastIndex = ranking.flatColumns.map((col) => (<any>col.desc).selectedId).lastIndexOf(id);
```

Comment:
`.flatColumns` is a quite expensive operation, you could search for the previous column directly instead of the index

Reply:
you mean using the `ranking.find(string | (Column) => boolean)` method? because it uses the flatColumns method too
 I thought of calling the `flatColumns` method only once and saving the result to a variable
what would be another method I could use?